crates/sqlexec/src/context.rs

```diff
@@ -87,6 +87,10 @@ pub struct SessionContext {
     env_reader: Option<Box<dyn EnvironmentReader>>,
     /// Job runner for background jobs.
     background_jobs: JobRunner,
+    /// Session's table providers.
+    table_providers: HashMap<uuid::Uuid, Arc<dyn TableProvider>>,
+    /// Session's physical plans.
+    physical_plans: HashMap<uuid::Uuid, Arc<dyn ExecutionPlan>>,
```

Comment:
Context getting chunky.

Are these specific to the "remote" side?

Reply:
Yeah, specific to remote side.

Reply:
Let's move these into a separate struct to avoid bloating up this Context struct with remote specific state.